torchgen/gen_functionalization_type.py

```diff
@@ -347,7 +347,6 @@ def emit_view_functionalization_body(
         }}
       );
       auto compute_reference_meta =
-        {view_tensor_name}.key_set().has_backend(c10::BackendComponent::XLABit) ||
```

Comment:
@bdhirsh Is this guard safe to remove? With it, I crashed on xla symbolic expand:
```
root@t1v-n-307ffe96-w-0:/workspaces/work/pytorch/xla# PJRT_DEVICE=CPU python test/test_dynamic_shapes.py -v TestDynamicShapes.test_simple_expand
test_simple_expand (__main__.TestDynamicShapes) ... ERROR

======================================================================
ERROR: test_simple_expand (__main__.TestDynamicShapes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_dynamic_shapes.py", line 18, in test_simple_expand
    t5.expand(t2.size(0))
RuntimeError: /workspaces/work/pytorch/build/aten/src/ATen/RegisterCompositeExplicitAutograd.cpp:2109: SymIntArrayRef expected to contain only concrete integers

----------------------------------------------------------------------
Ran 1 test in 0.022s

FAILED (errors=1)
```

Reply:
Okay, it regresses. It's not safe.

Reply:
@alanwaketan so, the idea motivating this bit if code is the following:
* pytorch/XLA doesn't care about strides, so when comparing a pytorch program when run on CUDA vs XLA, the user will witness different strides on the tensors throughout their program
* functionalization gives XLA the ability to fix that problem; XLA can choose to not care about the value of strides in all of its kernels, but functionalization can run the meta function for every ATen, to properly set the strides

One question here is - do you think that's a benefit worth trying to capture for pytorch/XLA (stride correctness, for the user's perspective, for XLA tensors)? I'd be interested in @JackCaoG 's opinion.

Our options are either to:
(1) kill that code, and not bother trying to get correct strides
(2) make it more robust so it works on this test

In this test, it looks like you're using dynamic shapes, and the meta function we're calling doesn't play well with dynamic shapes. The way that the dynamic shapes workstream in core has been handling this is that we have python implementations / decompositons of a bunch of our ops, that we want to run when dynamic shapes are enabled. And it looks like... for some reason we aren't calling that python impl, and are instead calling the C++ one?

There's probably a better way to arrange for this to work with XLA, but one option option is enable the python dispatcher in your test, which should override a bunch of C++ meta kernels with their python equivalents:
```
from torch._dispatch.python import enable_python_dispatcher
with enable_python_dispatcher():
    test()
```

Reply:
Here is the follow up on the xla side: https://github.com/pytorch/xla/issues/4448.

Reply:
Looks like by enabling python dispatcher and implementing missing sym size ops can workaround this. But then it brings a bigger question whether we should enable python dispatcher for dynamic shapes or not.